### PR TITLE
FIX: Issue #1885 error in generating roc curve for binary output feature

### LINKED ISSUE #1885: ludwig visualize roc_curves fail with KeyError with binary output feature
**Describe the bug**
When generating the ROC Curve with the `ludwig visualize` cli, the command fails with a "binary" output feature.  However, the command works if the output feature is defined as "category".  

With a binary output feature the failure is a `KeyError`.  Here is full stack trace:
```
Traceback (most recent call last):
  File "/usr/local/bin/ludwig", line 8, in <module>
    sys.exit(main())
  File "/usr/local/lib/python3.7/dist-packages/ludwig/cli.py", line 158, in main
    CLI()
  File "/usr/local/lib/python3.7/dist-packages/ludwig/cli.py", line 63, in __init__
    getattr(self, args.command)()
  File "/usr/local/lib/python3.7/dist-packages/ludwig/cli.py", line 98, in visualize
    visualize.cli(sys.argv[2:])
  File "/usr/local/lib/python3.7/dist-packages/ludwig/visualize.py", line 3823, in cli
    vis_func(**vars(args))
  File "/usr/local/lib/python3.7/dist-packages/ludwig/visualize.py", line 1022, in roc_curves_cli
    **kwargs,
  File "/usr/local/lib/python3.7/dist-packages/ludwig/visualize.py", line 3058, in roc_curves
    ground_truth = _vectorize_ground_truth(ground_truth, feature_metadata["str2idx"], ground_truth_apply_idx)
KeyError: 'str2idx'
```

**To Reproduce**

* First create this Ludwig config file, save with name `config.yaml`
```
preprocessing:
  number:
    normalization: zscore
    missing_value_strategy: fill_with_mean

input_features:
  - name: age
    type: number
  - name: workclass
    type: category
  - name: fnlwgt
    type: number
  - name: education
    type: category
  - name: education-num
    type: number
  - name: marital-status
    type: category
  - name: occupation
    type: category
  - name: relationship
    type: category
  - name: race
    type: category
  - name: sex
    type: category
  - name: capital-gain
    type: number
  - name: capital-loss
    type: number
  - name: hours-per-week
    type: number
  - name: native-country
    type: category

combiner:
  type: concat
  num_fc_layers: 3
  output_size: 128
  dropout: 0.2

output_features:
  - name: income
    type: binary
    preprocessing:
      fallback_true_label: " >50K"
    num_fc_layers: 4
    output_size: 32

trainer:
  epochs: 5
  optimizer: 
    type: sgd
```

Now run this sequence of cli commands:
```
ludwig datasets download adult_census_income

ludwig experiment --dataset adult_census_income.csv \
              --config config.yaml \
              --logging_level warning

ludwig visualize --visualization roc_curves \
                  --ground_truth adult_census_income.csv \
                  --split_file adult_census_income.split.csv \
                  --ground_truth_metadata results/experiment_run/model/training_set_metadata.json \
                  --probabilities results/experiment_run/predictions.parquet \
                  --output_feature_name income \
                  --output_directory visualizations \
                  --model_names Model1 \
                  --file_format png
```

`ludwig visualize` command should fail with cited error.

Please provide code, yaml config file and a sample of data in order to entirely reproduce the issue.
Issues that are not reproducible will be ignored.

**Expected behavior**
`ludwig visuzaliz` command should generate the ROC Curve

**Screenshots**

**Environment (please complete the following information):**

- OS: Mac OS, Docker Desktop
- Version MacOS 12.2.1, Docker Desktop 4.6.0
- Python version: 3.8
- Ludwig version 0.5rc2

**Additional context**
If the output feature is defined as a "category" output feature, the `ludwig visualize roc_curves` work as expected.

In looking at the visualize roc curve module, this appears to be the problematic statement: https://github.com/ludwig-ai/ludwig/blob/37048d7f9e39ec60e5488a2a99052d773e924dd5/ludwig/visualize.py#L3058 

This code is looking for the `str2idx` key in the output features metadata dictionary.  However, the metadata entry for a binary output feature looks like this
```
    "income": {
        "bool2str": [
            " <=50K",
            " >50K"
        ],
        "fallback_true_label": " >50K",
        "preprocessing": {
            "fallback_true_label": " >50K",
            "missing_value_strategy": "drop_row"
        },
        "str2bool": {
            " <=50K": false,
            " >50K": true
        }
    },
```
I'm attaching the full metadata file.
[training_set_metadata.json.txt](https://github.com/ludwig-ai/ludwig/files/8448139/training_set_metadata.json.txt)

### PR DESCRIPTION
**Pull request by jimthompson5802**
# Code Pull Requests

Fix  #1885 

Summary of changes:
* Updated roc curve visualization to account for difference in binary and category output features
* Add CLI test case for binary output feature

### CODE REVIEW

**Comment by jimthompson5802** [comment]
noticed same error in binary threshold plots.  Applied the same fix and updated unit test for binary threshold plots to test explicitly for binary output feature.

File: ludwig/visualize.py [hunk 1]
```diff
@@ -3055,7 +3074,24 @@ class and all the others will be considered negative. `positive_label` is
...
+                    ground_truth, feature_metadata["str2bool"], ground_truth_apply_idx
+                )
+            else:
+                # standard boolean representation
+                ground_truth = ground_truth.values
+
+            # convert to 0/1 representation
+            ground_truth = ground_truth.astype(np.int)
+            # ensure positive_label is 1 for binary feature
+            positive_label = 1
```
Comment (justinxzhao) [inline ludwig/visualize.py:3094]: This code looks very similar / duplicated as the one from above. Could we factor this out into a separate method and call it in both places?

Comment (jimthompson5802) [reply ludwig/visualize.py:3094]: got it...wilco.

Comment (jimthompson5802) [reply ludwig/visualize.py]: @justinxzhao factored out common code into a callable function.

**Pull request merged**